python_modules/libraries/dagster-components/dagster_components/utils.py

```diff
@@ -134,83 +99,41 @@ def get_asset_spec(self, base_spec: AssetSpec, x_params: Any) -> AssetSpec:
             model=self.asset_attributes,
             context=self.resolution_context.with_scope(**context),
         )
-        return base_spec.replace_attributes(**resolved_attributes)
+        if "code_version" in resolved_attributes:
+            resolved_attributes = {
+                **resolved_attributes,
+                "code_version": str(resolved_attributes["code_version"]),
+            }
+        return base_spec.replace_attributes(
+            **{k: v for k, v in resolved_attributes.items() if k not in TRANSLATOR_MERGE_ATTRIBUTES}
+        ).merge_attributes(
+            **{k: v for k, v in resolved_attributes.items() if k in TRANSLATOR_MERGE_ATTRIBUTES}
+        )
 
 
 def get_wrapped_translator_class(translator_type: type):
     """Temporary hack to allow wrapping of many methods of a given translator class. Will be removed
     once all translators implement `get_asset_spec`.
     """
+    check.invariant(
+        hasattr(translator_type, "get_asset_spec"),
+        "Translator must implement get_asset_spec",
+    )
```

Comment:
would probably prefer to just delete `get_wrapped_translator_class` and have  the WrappedSlingTranslator do this manually, and make whatever code is required to do that as simple as possible.

Do we even need the `_specs_map` bit? or could we just `cached_method` this stuff?

Reply:
not sure we can `cached_method` since the input is not generally hashable